cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -805,6 +805,32 @@ private void configProviders(KafkaConfiguration userConfig)    {
 
         writer.println();
     }
+    
+    /**
+     * Get the user provided Kafka configuration provider aliases, throwing an InvalidConfigurationException if any are found that would overwrite the Strimzi defined configuration providers
+     * 
+     * @param strimziConfigProviders    The Strimzi defined configuration providers
+     * @param userConfig                The user configuration to extract the possible user-provided config provider configuration from
+     * @return                          The user defined Kafka configuration provider aliases or empty string
+     */
+    private String getUserConfigProviderAliases(String strimziConfigProviders, KafkaConfiguration userConfig) {
+        String userConfigProviderAliases = "";
+        if (userConfig != null
+                && !userConfig.getConfiguration().isEmpty()
+                && userConfig.getConfigOption("config.providers") != null) {
+            userConfigProviderAliases = userConfig.getConfigOption("config.providers");
+            Collection<String> userAliases = Arrays.asList(userConfigProviderAliases.split(","));
+            Collection<String> strimziAliases = Arrays.asList(strimziConfigProviders.split(","));
+            
+            userAliases.stream().forEach(alias -> {
+                if (strimziAliases.contains(alias)) {
+                    throw new InvalidConfigurationException("config.provider " + alias + " not permitted as it reserved for Strimzi. Not permitted aliases: " + strimziAliases); 
+                }
+            });
+            userConfig.removeConfigOption("config.providers");
+        }
+        return userConfigProviderAliases;
```

Comment:
I guess this can have a bit improved readability with something like this?

```suggestion
        if (userConfig != null
                && !userConfig.getConfiguration().isEmpty()
                && userConfig.getConfigOption("config.providers") != null) {
            String userConfigProviderAliases = userConfig.getConfigOption("config.providers");
            Collection<String> userAliases = Arrays.asList(userConfigProviderAliases.split(","));
            Collection<String> strimziAliases = Arrays.asList(strimziConfigProviders.split(","));
            
            userAliases.stream().forEach(alias -> {
                if (strimziAliases.contains(alias)) {
                    throw new InvalidConfigurationException("config.provider " + alias + " not permitted as it reserved for Strimzi. Not permitted aliases: " + strimziAliases); 
                }
            });
            
            userConfig.removeConfigOption("config.providers");
            return userConfigProviderAliases;
        } else {
            return "";
        }
```